### PR TITLE
[v5] Final `.data` to `.output`

### PR DESCRIPTION
**Pull request by davidkpiano**
This PR renames `.data` to `.output` on final states:

```diff
const machine = createMachine({
  // ...
  states: {
    // ...
    success: {
-     data: { message: 'Success!' }
+     output: { message: 'Success!' }
    }
  }
})
```

### CODE REVIEW

File: .changeset/five-mirrors-join.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+---
+'xstate': major
+'@xstate/fsm': major
+'@xstate/react': major
+'@xstate/vue': major
```
Comment (Andarist) [inline .changeset/five-mirrors-join.md:5]: I don't think that this change affects any of those libraries.
```suggestion
```

File: packages/core/src/StateNode.ts [hunk 2]
```diff
@@ -187,7 +187,7 @@ export class StateNode<
     this.meta = this.config.meta;
     this.doneData =
```
Comment (Andarist) [inline packages/core/src/StateNode.ts:188]: this one could be renamed as well but we can also do it later

File: .changeset/five-mirrors-join.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+The output data on final states is now specified as `.output` instead of `.data`:
+
+```diff
+const machine = createMachine({
+  // ...
+  states: {
+    // ...
+    success: {
+-     data: { message: 'Success!' }
++     output: { message: 'Success!' }
```
Comment (Andarist) [inline .changeset/five-mirrors-join.md:17]: this later becomes part of the `done.state.*` and `done.invoe.*` events, should we rename the properties in those events as well?

Comment (davidkpiano) [reply .changeset/five-mirrors-join.md:14]: I'm not sure. I think there should be a difference between "this is the output of this machine" and "this is the data in this event". Not sure yet.

**Pull request merged**